.github/workflows/release-module-npm.yml

```diff
@@ -46,15 +52,15 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
       - uses: actions/setup-node@v4
         with:
-          node-version: '20.x'
+          node-version: ${{ inputs.node_version }}
           registry-url: ${{ inputs.registry_url }}
           scope: ${{ inputs.registry_scope }}
       - name: Install Packages
         run: npm ci
       - name: Build
-        run: npm run build
-        if: inputs.npm_build == true
-      - name: Publish to npmjs packages
+        run: ${{ inputs.npm_build_command }}
+        if: inputs.npm_build_command != ''
+      - name: Publish to ${{ inputs.registry_url }}
         run: npm publish --access ${{   inputs.package_access }}
         env:
-          NODE_AUTH_TOKEN: ${{ secrets.REGISTRY_AUTH_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ inputs.registry_url == 'https://npm.pkg.github.com' && secrets.GITHUB_TOKEN || secrets.REGISTRY_AUTH_TOKEN }}
```

Comment:
I think we should have a couple of commands here with some `if` logic, this way we don't have to update multiple other workflows that are currently being used.

Reply:
That is what this does

Reply:
Is the desired behaviour that you will have two npm publish commands run?  Because the second command does not have a `if` statement

Reply:
ah wait, its a build not a publish ,ignore.

Reply:
There's only one publish command. The alternative would be to take the approach we have with docker where pass booleans for `npm` and `gh npm` and then have two publish commands. That might be better but doesn't scale to other registries if ever needed.

It's have to ever reliably share workflows for the two publishes though because the registry is setup in the node setup step.